realm/realm-library/src/main/java/io/realm/SetValueOperator.java

```diff
@@ -340,17 +400,12 @@ boolean containsInternal(@Nullable Object o) {
     }
 
     @Override
-    boolean remove(@Nullable Object o) {
+    boolean removeInternal(@Nullable Object o) {
         return osSet.remove((Long) o);
     }
 
     @Override
-    boolean containsAll(Collection<?> c) {
-        if (isRealmCollection(c)) {
-            OsSet otherOsSet = ((RealmSet<?>) c).getOsSet();
-            return funnelCollection(otherOsSet, OsSet.ExternalCollectionOperation.CONTAINS_ALL);
-        }
-
+    boolean containsAllInternal(Collection<?> c) {
         for (Object value : c) {
```

Comment:
Not you can just use `isCollectionSameType()`.

Reply:
The logic now works as follows:
call to operation: perform type check on collection/object -> operation internal implementation: collection is type safe.

This applies to all operations now.

Reply:
I.e. all "internal" methods are safe to execute since the logic for type-checking is similar for all value operators and is performed in the abstract, parent value operator.